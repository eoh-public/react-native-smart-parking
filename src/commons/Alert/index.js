import React from 'react';
import Modal from 'react-native-modal';
import { View, StyleSheet } from 'react-native';

import Text from '../Text';

import { Colors } from '../../configs';
import TextButton from '../TextButton';

class Alert extends React.Component {
  static _ref = null;

  static setRef(ref = {}) {
    this._ref = ref;
  }

  static getRef() {
    return this._ref;
  }

  static clearRef() {
    this._ref = null;
  }

  state = {
    title: '',
    description: '',
    buttons: [],
    show: false,
    styleButtons: [] || {},
  };

  static alert = (title, description, buttons, styleButtons) => {
    this._ref.setState({
      title,
      description,
      buttons,
      show: true,
      styleButtons,
    });
  };

  renderContent = () => {
    const { description, title } = this.state;
    return (
      <>
        <Text style={styles.title}>{title}</Text>
        <View style={styles.descriptionWrap}>
          {React.isValidElement(description) ? (
            description
          ) : (
            <Text style={styles.description}>{description}</Text>
          )}
        </View>
      </>
    );
  };

  renderButtons = () => {
    const justifyContent =
      this.state.buttons.length === 1 ? 'center' : 'space-between';
    return (
      <View
        key={this.state.title}
        style={[
          styles.flexRow,
          {
            justifyContent: justifyContent,
          },
          this.state.styleButtons,
        ]}
      >
        {this.state.buttons.map((item, index) => {
          return (
            <TextButton
              key={`item_${index}`}
              primary={item.primary}
              title={item.label}
              onPress={item.onPress}
              wrapStyle={item.style}
            />
          );
        })}
      </View>
    );
  };

  static hideModal = () => {
    this._ref.setState({ show: false });
  };

  static showModal = () => {
    this._ref.setState({ show: true });
  };

  render() {
    const { buttons, show } = this.state;
    return (
      <Modal animationIn={'zoomIn'} animationOut={'zoomOut'} isVisible={show}>
        <View style={[styles.wrap, styles.contentPadding]}>
          {this.renderContent()}
        </View>
        <View style={[styles.wrap, styles.buttonsPadding]}>
          {!!buttons?.length && this.renderButtons()}
        </View>
      </Modal>
    );
  }
}

const styles = StyleSheet.create({
  wrap: {
    padding: 20,
    alignItems: 'center',
    backgroundColor: Colors.White,
  },
  title: {
    fontWeight: 'bold',
    fontSize: 18,
    color: Colors.Primary,
  },
  descriptionWrap: {
    minHeight: 78,
    paddingVertical: 10,
  },
  description: {
    textAlign: 'center',
  },
  flexRow: {
    flexDirection: 'row',
  },
  contentPadding: {
    paddingBottom: 0,
  },
  buttonsPadding: {
    paddingTop: 0,
    alignItems: 'stretch',
  },
});

export default Alert;
