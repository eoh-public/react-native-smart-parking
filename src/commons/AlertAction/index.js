import React from 'react';
import { StyleSheet, View } from 'react-native';
import Modal from 'react-native-modal';

import Text from '../../commons/Text';
import ViewButtonBottom from '../ViewButtonBottom';
import { Colors, Device } from '../../configs';

const AlertAction = ({
  visible,
  hideModal,
  title,
  message,
  leftButtonTitle,
  leftButtonClick,
  rightButtonTitle,
  rightButtonClick,
  rightButtonStyle,
  children,
  testIDPrefix = '',
}) => {
  return (
    <Modal
      isVisible={visible}
      onBackButtonPress={hideModal}
      onBackdropPress={hideModal}
      style={styles.container}
    >
      <View style={styles.popoverStyle}>
        <View style={styles.modalWrapper}>
          <View style={styles.modalHeader}>
            <Text semibold style={styles.modalHeaderText}>
              {title}
            </Text>
          </View>
          {!!message && (
            <Text style={[styles.descriptionText, styles.descriptionText]}>
              {message}
            </Text>
          )}
          {children}
          <ViewButtonBottom
            leftTitle={leftButtonTitle}
            onLeftClick={leftButtonClick}
            rightTitle={rightButtonTitle}
            onRightClick={rightButtonClick}
            styleButtonRightText={rightButtonStyle}
            testIDPrefix={testIDPrefix}
          />
        </View>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 0,
  },
  popoverStyle: {
    width: '100%',
    backgroundColor: Colors.White,
    bottom: 0,
    left: 0,
    position: 'absolute',
    borderRadius: 10,
  },
  modalWrapper: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: Colors.White,
    borderRadius: 10,
    marginBottom: Device.isIphoneX ? 20 : 0,
  },
  modalHeader: {
    padding: 16,
    backgroundColor: Colors.White,
    borderBottomWidth: 1,
    borderTopLeftRadius: 20,
    borderTopRightRadius: 20,
    borderColor: Colors.Gray4,
  },
  modalHeaderText: {
    fontSize: 16,
    lineHeight: 24,
    color: Colors.Gray9,
  },
  descriptionText: {
    paddingHorizontal: 16,
    paddingTop: 16,
  },
});

export default AlertAction;
