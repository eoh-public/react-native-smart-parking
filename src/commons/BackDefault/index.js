import React, { memo } from 'react';
import { TouchableOpacity, StyleSheet } from 'react-native';
import { IconOutline } from '@ant-design/icons-react-native';
import { TESTID } from '../../configs/Constants';

const BackDefault = memo(({ goBack, color, fixedHeight }) => {
  return (
    <TouchableOpacity
      testID={TESTID.BACK_DEFAULT_TOUCH}
      onPress={goBack}
      style={[styles.wrap, fixedHeight && styles.noPaddingVertical]}
    >
      <IconOutline name="left" size={27} color={color} />
    </TouchableOpacity>
  );
});

const styles = StyleSheet.create({
  wrap: {
    paddingLeft: 12,
    paddingVertical: 14,
  },
  noPaddingVertical: {
    paddingVertical: 0,
  },
});

export default BackDefault;
