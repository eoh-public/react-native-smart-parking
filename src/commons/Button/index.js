import React from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';

import { Colors } from '../../configs';
import Text from '../Text';

import withPreventDoubleClick from '../WithPreventDoubleClick';

const ButtonStyle = {
  auth: {
    backgroundColor: Colors.Primary,
    borderRadius: 5,
    borderWidth: 0,
  },
  primary: {
    backgroundColor: Colors.Primary,
    borderRadius: 28,
    borderWidth: 0,
  },
  info: {
    backgroundColor: Colors.White,
    borderRadius: 28,
    borderWidth: 1,
    borderColor: Colors.Gray4,
  },
  cancel: {
    backgroundColor: Colors.White,
    borderRadius: 28,
    borderWidth: 1,
    borderColor: Colors.Gray4,
  },
  disabled: {
    backgroundColor: Colors.Gray4,
    borderRadius: 28,
    borderWidth: 0,
  },
  disabledBorder: {
    backgroundColor: Colors.Gray4,
    borderRadius: 28,
    borderWidth: 1,
    borderColor: Colors.Gray3,
  },
  primaryBorder: {
    backgroundColor: Colors.Cyan1,
    borderRadius: 28,
    borderWidth: 1,
    borderColor: Colors.Primary,
  },
  alert: {
    backgroundColor: Colors.Red6,
    borderRadius: 28,
    borderWidth: 0,
  },
  alertBorder: {
    backgroundColor: Colors.White,
    borderRadius: 28,
    borderWidth: 1,
    borderColor: Colors.Red6,
  },
  underline: {
    backgroundColor: Colors.White,
    borderWidth: 0,
  },
  setupBorder: {
    backgroundColor: Colors.Cyan1,
    borderRadius: 28,
    borderWidth: 1,
    borderColor: Colors.Primary,
  },
};

const TextColor = {
  auth: Colors.White,
  primary: Colors.White,
  info: Colors.Primary,
  cancel: Colors.Gray8,
  disabled: Colors.Gray6,
  disabledBorder: Colors.Gray6,
  primaryBorder: Colors.Primary,
  alert: Colors.White,
  alertBorder: Colors.Red6,
  underline: Colors.Gray6,
  setupBorder: Colors.Primary,
};

const TextSize = {
  H1: 30,
  H2: 24,
  H3: 20,
  H4: 16,
  Body: 14,
  Label: 12,
};

// Note: pass props style with flex: 0 if background does not display

// Type
// auth, primary, info, cancel, disabled, disabledBorder, primaryBorder, alert, alertBorder, underline, setupBorder

export default ({
  title,
  onPress,
  width,
  height = 48,
  activeOpacity,
  type,
  icon,
  textType = 'H4',
  textSemiBold = true,
  style,
  testID,
}) => {
  const styleButton = ButtonStyle[type];
  const textColor = TextColor[type];

  const isDisabled = type === 'disabled' || type === 'disabledBorder';
  const isUnderline = type === 'underline';
  const textSize = TextSize[textType];
  const PreventDoubleTouch = withPreventDoubleClick(TouchableOpacity);
  return (
    <PreventDoubleTouch
      testID={testID}
      style={[
        {
          width,
          height,
        },
        styles.button,
        styleButton,
        style,
      ]}
      onPress={onPress}
      disabled={isDisabled}
      activeOpacity={activeOpacity}
    >
      <View style={styles.wrap}>
        {icon}
        <Text
          type={textType}
          semibold={textSemiBold}
          color={textColor}
          size={textSize}
          underline={isUnderline}
          style={[icon && styles.marginIcon]}
        >
          {title}
        </Text>
      </View>
    </PreventDoubleTouch>
  );
};

const styles = StyleSheet.create({
  button: {
    width: '100%',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.Primary,
  },
  marginIcon: {
    marginLeft: 10,
    marginBottom: -3,
  },
  wrap: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
});
