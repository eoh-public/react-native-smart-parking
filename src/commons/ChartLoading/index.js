import React from 'react';
import Svg, { Text } from 'react-native-svg';
import t from 'i18n';

const ChartLoading = ({ message }) => {
  return (
    <Svg width={'100%'} height={300}>
      <Text
        x="50%"
        y="47%"
        alignment-baseline="middle"
        textAnchor="middle"
        fontSize="20"
        fontWeight="bold"
        fill="black"
      >
        {message || t('loading')}
      </Text>
    </Svg>
  );
};
export default ChartLoading;
