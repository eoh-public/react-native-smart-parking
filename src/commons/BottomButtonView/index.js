import React, { memo } from 'react';
import { View, StyleSheet } from 'react-native';
import { Colors } from '../../configs';

import { TESTID } from '../../configs/Constants';
import Button from '../Button';

const BottomButtonView = memo(
  ({
    rowButton = false,
    mainIcon,
    mainTitle,
    secondaryTitle,
    style,
    onPressMain,
    onPressSecondary,
    topComponent,
    semiboldMain,
    semiboldSecond,
    textTypeMain = 'H4',
    typeMain = 'primary',
    typeSecondary = 'cancel',
    testIDPrefix = '',
    disabled = false,
  }) => {
    return (
      <View
        style={[
          styleCustom.container,
          rowButton && styleCustom.horizontalContainer,
          style,
        ]}
      >
        {topComponent}
        {mainTitle && (
          <Button
            icon={mainIcon}
            type={disabled ? 'disabled' : typeMain}
            title={mainTitle}
            onPress={onPressMain}
            textSemiBold={semiboldMain}
            textType={textTypeMain}
            style={rowButton && styleCustom.buttonMainRow}
            testID={`${testIDPrefix}${TESTID.BOTTOM_VIEW_MAIN}`}
          />
        )}
        {secondaryTitle && (
          <Button
            type={disabled ? 'disabled' : typeSecondary}
            title={secondaryTitle}
            onPress={onPressSecondary}
            textSemiBold={semiboldSecond}
            style={
              rowButton
                ? styleCustom.buttonSecondaryRow
                : styleCustom.buttonSecondaryColumn
            }
            testID={`${testIDPrefix}${TESTID.BOTTOM_VIEW_SECONDARY}`}
          />
        )}
      </View>
    );
  }
);

export default BottomButtonView;

const styleCustom = StyleSheet.create({
  horizontalContainer: {
    flexDirection: 'row-reverse',
    justifyContent: 'center',
  },
  container: {
    paddingHorizontal: 16,
    paddingVertical: 20,
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    flex: 1,
    borderTopWidth: 1,
    borderTopColor: Colors.Gray4,
    backgroundColor: Colors.White,
  },
  mainIcon: {
    marginRight: 10,
  },
  buttonMainRow: {
    marginLeft: 8,
  },
  buttonSecondaryRow: {
    marginRight: 8,
  },
  buttonSecondaryColumn: {
    marginTop: 8,
  },
});
