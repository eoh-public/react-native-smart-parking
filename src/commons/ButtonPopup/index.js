import React from 'react';
import { StyleSheet, View } from 'react-native';
import Modal from 'react-native-modal';
import { Icon } from '@ant-design/react-native';

import { Colors, Device } from '../../configs';
import { TESTID } from '../../configs/Constants';
import Text from '../../commons/Text';

import BottomButtonView from '../BottomButtonView';

const ButtonPopup = ({
  rowButton = true,
  visible,
  children,
  mainTitle,
  secondaryTitle,
  thirdTitle,
  onClose,
  onPressMain,
  onPressSecondary,
  onPressThird,
  bodyStyle,
  hideClose,
  typeMain,
  typeSecondary,
  typeThird,
  semiboldSecond,
  title,
  childrenStyle,
  titleStyle,
  bottomStyles,
}) => {
  return (
    <Modal
      isVisible={visible}
      onBackButtonPress={onClose}
      onBackdropPress={onClose}
      testID={TESTID.MODAL_BUTTON_POPUP}
    >
      <View style={styles.popoverStyle}>
        <View style={styles.modalWrapper}>
          <View style={titleStyle}>
            {title && (
              <Text type="H4" bold>
                {title}
              </Text>
            )}
            {!hideClose && (
              <Icon
                name={'close'}
                color={Colors.Gray8}
                style={styles.close}
                onPress={onClose}
              />
            )}
          </View>
          <View
            style={[
              childrenStyle ? childrenStyle : styles.modalHeader,
              bodyStyle,
            ]}
          >
            {children}
          </View>
          <BottomButtonView
            rowButton={rowButton}
            style={[
              styles.bottomButtonView,
              thirdTitle && styles.thirdStyle,
              bottomStyles,
            ]}
            secondaryTitle={secondaryTitle}
            mainTitle={mainTitle}
            onPressSecondary={onPressSecondary}
            onPressMain={onPressMain}
            typeMain={typeMain}
            typeSecondary={typeSecondary}
            testIDPrefix={TESTID.PREFIX.BUTTON_POPUP}
            semiboldSecond={semiboldSecond}
          />
          {thirdTitle && (
            <BottomButtonView
              style={styles.bottomButtonView2}
              mainTitle={thirdTitle}
              onPressMain={onPressThird}
              typeMain={typeThird}
            />
          )}
        </View>
      </View>
    </Modal>
  );
};

export default ButtonPopup;

const styles = StyleSheet.create({
  popoverStyle: {
    width: '100%',
    backgroundColor: Colors.White,
    position: 'absolute',
    borderRadius: 10,
  },
  modalWrapper: {
    flex: 1,
    flexDirection: 'column',
    position: 'relative',
    backgroundColor: Colors.White,
    borderRadius: 10,
    marginBottom: Device.isIphoneX ? 20 : 0,
  },
  modalHeader: {
    padding: 16,
    paddingTop: 60,
  },
  close: {
    position: 'absolute',
    top: 20,
    right: 20,
    zIndex: 1,
  },
  bottomButtonView: {
    position: 'relative',
    marginBottom: 24,
  },
  bottomButtonView2: {
    position: 'relative',
    marginBottom: 24,
  },
  thirdStyle: {
    marginBottom: 0,
  },
});
