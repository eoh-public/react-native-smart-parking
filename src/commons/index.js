import { CircleButton } from './CircleButton';
import FullLoading from './FullLoading';
import ButtonPopup from './ButtonPopup';
import { CircleView } from './CircleView';
import AlertAction from './AlertAction';
import CustomCheckbox from './CustomCheckbox';
import ImagePicker from './ImagePicker';
import RadioCircle from './RadioCircle';
import ExpandView from './ExpandView';
import TouchableScale from './TouchableScale';
import { Section } from './Section';
import ViewButtonBottom from './ViewButtonBottom';
import Today from './Today';
import MenuActionList from './MenuActionList';
import Button from './Button';
import Alert from './Alert';
import MenuActionMore from './MenuActionMore';
import ImageButton from './ImageButton';
import MenuActionAddnew from './MenuActionAddnew';

export {
  CircleButton,
  FullLoading,
  ButtonPopup,
  CircleView,
  AlertAction,
  CustomCheckbox,
  ImagePicker,
  RadioCircle,
  ExpandView,
  TouchableScale,
  Section,
  ViewButtonBottom,
  Today,
  MenuActionList,
  Button,
  Alert,
  MenuActionMore,
  ImageButton,
  MenuActionAddnew
};
