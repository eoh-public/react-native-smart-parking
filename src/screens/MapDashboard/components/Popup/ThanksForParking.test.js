import ThanksForParkingPopup from './ThanksForParking';
import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { ButtonPopup } from '../../../../commons';

describe('Test ThanksForParkingPopup', () => {
  let tree;

  test('render ThanksForParkingPopup', () => {
    const onClose = jest.fn();
    act(() => {
      tree = renderer.create(
        <ThanksForParkingPopup visible={true} onClose={onClose} />
      );
    });
    const instance = tree.root;
    const popup = instance.findAllByType(ButtonPopup);
    expect(popup.length).toBe(1);
  });
});
