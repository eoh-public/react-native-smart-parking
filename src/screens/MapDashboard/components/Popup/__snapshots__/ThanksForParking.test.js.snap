// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ThanksForParkingPopup render ThanksForParkingPopup 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  testID="MODAL_BUTTON_POPUP"
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "flex": 1,
            "flexDirection": "column",
            "marginBottom": 20,
            "position": "relative",
          }
        }
      >
        <View>
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "#595959",
                  "fontFamily": "antoutline",
                  "fontSize": 22,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {
                  "position": "absolute",
                  "right": 20,
                  "top": 20,
                  "zIndex": 1,
                },
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "padding": 16,
                "paddingTop": 60,
              },
              Object {
                "alignItems": "center",
                "flex": 1,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "antoutline",
                  "fontSize": 42,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {
                  "color": "#00979D",
                  "marginBottom": 19,
                },
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "SFProDisplay-Semibold",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {
                  "marginBottom": 8,
                },
              ]
            }
          >
            Đỗ xe hoàn tất!
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "textAlign": "left",
                  "textDecorationLine": "none",
                },
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            Vui lòng dời xe của bạn khỏi vị trí đỗ.
Cảm ơn đã sử dụng dịch vụ đỗ xe SmartParking.
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "flex": 1,
                "left": 0,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
                "position": "absolute",
                "right": 0,
              },
              Object {
                "flexDirection": "row-reverse",
                "justifyContent": "center",
              },
              Array [
                Object {
                  "marginBottom": 24,
                  "position": "relative",
                },
                undefined,
              ],
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00979D",
                "borderRadius": 28,
                "borderWidth": 0,
                "flex": 1,
                "height": 48,
                "justifyContent": "center",
                "marginLeft": 4,
                "opacity": 1,
                "width": "100%",
              }
            }
            testID="BUTTON_POPUPBOTTOM_VIEW_MAIN"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "SFProDisplay-Semibold",
                      "fontSize": 16,
                      "lineHeight": 24,
                      "textAlign": "left",
                      "textDecorationLine": "none",
                    },
                    Object {},
                  ]
                }
              >
                Xong
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
